cpp/src/arrow/pretty_print-test.cc

```diff
@@ -144,11 +202,72 @@ TEST_F(TestPrettyPrint, DictionaryType) {
   auto arr = std::make_shared<DictionaryArray>(dict_type, indices);
 
   static const char* expected = R"expected(
--- is_valid: [true, true, false, true, true, true]
--- dictionary: ["foo", "bar", "baz"]
--- indices: [1, 2, null, 0, 2, 0])expected";
+-- is_valid:
+  [
+    true,
+    true,
+    false,
+    true,
+    true,
+    true
+  ]
```

Comment:
Seeing this again it seems like the `is_valid` is redundant since the nulls appear in the indices

Reply:
Yes, this should go.